src/pv.rs

```diff
@@ -61,6 +62,62 @@ impl PvPacket {
             private: std::ptr::null_mut(),
         }
     }
+
+    // check whether the new header can be attached to payload from the start of packet
+    fn is_addable(header_length: usize, packet: &PvPacket) -> bool {
+        let total_len: u32 = packet.size - packet.start;
+        let payload_len: u32 = packet.end - packet.start;
+
+        (header_length as u32) + payload_len <= total_len
+    }
+
+    // add header to payload
+    pub fn add_header(&mut self, header: Vec<u8>) -> Result<(), ()> {
```

Comment:
attach or prepend

Reply:
The name is changed with `attach_header()`